examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/ObserverModeBillingClient.kt

```diff
@@ -0,0 +1,61 @@
+package com.revenuecat.purchasetester
+
+import android.app.Activity
+import android.content.Context
+import com.android.billingclient.api.BillingClient
+import com.android.billingclient.api.BillingClientStateListener
+import com.android.billingclient.api.BillingFlowParams
+import com.android.billingclient.api.BillingResult
+import com.android.billingclient.api.ProductDetails
+import com.android.billingclient.api.Purchase
+import com.android.billingclient.api.PurchasesUpdatedListener
+import com.revenuecat.purchases.Purchases
+
+object ObserverModeBillingClient : PurchasesUpdatedListener, BillingClientStateListener {
+    private lateinit var billingClient: BillingClient
+
+    fun start(context: Context) {
+        billingClient = BillingClient.newBuilder(context)
+            .setListener(this)
+            .enablePendingPurchases()
+            .build()
+        billingClient.startConnection(this)
+    }
+
+    fun purchase(
+        activity: Activity,
+        productDetails: ProductDetails,
+        offerToken: String?,
+        isOfferPersonalized: Boolean = false
+    ) {
+        val flowParams = BillingFlowParams.newBuilder()
+            .setProductDetailsParamsList(
+                listOf(
+                    BillingFlowParams.ProductDetailsParams.newBuilder()
+                        .setProductDetails(productDetails)
+                        .also { params ->
+                            offerToken?.let {
+                                params.setOfferToken(it)
+                            }
+                        }
+                        .build()
+                )
+            )
+            .setIsOfferPersonalized(isOfferPersonalized)
+            .build()
+        billingClient.launchBillingFlow(activity, flowParams)
+    }
+
+    override fun onBillingSetupFinished(billingResult: BillingResult) {
+        print("onBillingSetupFinished")
+    }
+    override fun onBillingServiceDisconnected() {
+        print("onBillingServiceDisconnected")
```

Comment:
We should use the `TesterLogHandler` in all these so the prints appear in the logs screen (which isn't very usable right now but I hope we can iterate on)

Reply:
Added! 💪